lib/route/link/bridge_info.c

```diff
@@ -105,6 +145,159 @@ static struct rtnl_link_info_ops bridge_info_ops = {
 		return -NLE_OPNOTSUPP; \
 	}
 
+/**
+ * Set VLAN filtering flag
+ * @arg link		Link object of type bridge
+ * @arg vlan_filtering	VLAN_filtering boolean flag to set.
+ *
+ * @see rtnl_link_bridge_get_vlan_filtering()
+ *
+ * @return Zero on success, otherwise a negative error code.
+ * @retval -NLE_OPNOTSUPP Link is not a bridge
```

Comment:
actually, `-NLE_OPNOTSUPP` only gets returned by an assertion failure with `IS_BRIDGE_INFO_ASSERT()`. That is not a regular error that can happen, because it's a bug of the caller to call the function this way.

So don't document this as a possible return code.  In practice, this function cannot fail...

At which point, it begs the question why this function ever would fail. I mean, you can all it with an invalid argument, like a dangling pointer or NULL (don't do that!!), but otherwise it doesn't fail.  Maybe such functions really should just return `void`, because this suggests to the caller that they should check for errors. Such error checking is cumbersome but also doing something that cannot happen.

Reply:
I changed all setter functions to return a void and now `#define IS_BRIDGE_INFO_ASSERT(link)` doesn't have `return -NLE_OPNOTSUPP`.